src/Evolution/DiscontinuousGalerkin/SlopeLimiters/WenoGridHelpers.cpp

```diff
@@ -5,44 +5,97 @@
 
 #include "DataStructures/Tensor/Tensor.hpp"
 #include "Domain/Direction.hpp"  // IWYU pragma: keep
+#include "Domain/Element.hpp"    // IWYU pragma: keep
 #include "Domain/LogicalCoordinates.hpp"
 #include "Domain/Mesh.hpp"  // IWYU pragma: keep
 #include "Domain/Side.hpp"
+#include "ErrorHandling/Assert.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
 
+namespace {
+
+template <size_t VolumeDim>
+bool check_element_has_one_similar_neighbor_in_direction(
+    const Element<VolumeDim>& element,
+    const Direction<VolumeDim>& direction) noexcept {
+  const auto& neighbors = element.neighbors().at(direction);
+  if (neighbors.size() > 1) {
+    // More than one neighbor
+    return false;
+  } else {
+    const auto& orientation_map = neighbors.orientation();
+    const auto& neighbor_segment_ids = neighbors.ids().cbegin()->segment_ids();
+    const auto reoriented_neighbor_segment_ids =
+        orientation_map.inverse_map()(neighbor_segment_ids);
+    if (reoriented_neighbor_segment_ids != element.id().segment_ids()) {
```

Comment:
I don't see how this can be correct; don't you want to check the refinement levels in each dimension?

Reply:
Whoops, that is indeed what I intended to do :(